packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1957,6 +2013,39 @@ export class TransactionController extends BaseController<
       this.updateTransaction.bind(this),
     );
   }
+
+  private async signTransaction(
+    transactionMeta: TransactionMeta,
+  ): Promise<string | undefined> {
+    const { txParams } = transactionMeta;
+
+    const unsignedEthTx = this.prepareUnsignedEthTx(txParams);
+    this.inProcessOfSigning.add(transactionMeta.id);
+    const signedTx = await this.sign?.(
+      unsignedEthTx,
+      txParams.from,
+      ...this.getAdditionalSignArguments(transactionMeta),
+    );
+    if (!signedTx || !this.afterSign(transactionMeta, signedTx)) {
+      log('Skipping signed status based on hook');
```

Comment:
Sorry for the confusion, I meant could we have two conditions here to provide alternate messages. Otherwise it will always claim it was the hook, when it could have been the keyring returning no signed transaction.

Something like:

```
if(!signedTx) {
    log('Skipping signed status as no signed transaction');
    return;
}

if(!this.afterSign(transactionMeta, signedTx)) {
    log('Skipping signed status based on hook');
    return;
}
```

Reply:
ow my bad, I split the conditions now.